code/addons/test/src/postinstall.ts

```diff
@@ -51,57 +55,158 @@ export default async function postInstall(options: PostinstallOptions) {
       ? info.rendererPackageName
       : null;
 
-  if (!annotationsImport) {
-    logger.info('The Vitest addon cannot yet be used with: ' + info.frameworkPackageName);
+  const prerequisiteCheck = async () => {
+    const reasons = [];
+
+    if (
+      info.frameworkPackageName !== '@storybook/nextjs' &&
+      info.builderPackageName !== '@storybook/builder-vite'
+    ) {
+      reasons.push(
+        'The Vitest addon can only be used with a Vite-based Storybook framework or Next.js.'
+      );
+    }
+
+    if (!annotationsImport) {
+      reasons.push(dedent`
+        The Vitest addon cannot yet be used with ${colors.pink.bold(info.frameworkPackageName)}
+      `);
+    }
+
+    const vitestVersion = await packageManager.getInstalledVersion('vitest');
+    if (vitestVersion && !satisfies(vitestVersion, '>=2.0.0')) {
+      reasons.push(`
+        The Vitest addon requires Vitest 2.0.0 or later.
+        Please update your ${colors.pink.bold('vitest')} dependency and try again.
+      `);
+    } else {
+      const depRanges = await packageManager.getAllDependencies();
+      const checkDependencies = [...dependencies, ...optionalDependencies];
+      const latestVersions = await packageManager.getVersions(...checkDependencies);
+      const latestPackages = checkDependencies.map(
+        (pkg, index) => [pkg, latestVersions[index].replace('^', '')] as const
+      );
+
+      let hasInconsistentPackageVersions = false;
+      for (const [pkg, latestVersion] of latestPackages) {
+        if (depRanges[pkg]) {
+          if (!dependencies.includes(pkg)) {
+            // Add found optional dependency so it will be updated to the latest version
+            dependencies.push(pkg);
+          }
+
+          if (!satisfies(latestVersion, depRanges[pkg])) {
+            hasInconsistentPackageVersions = true;
+            reasons.push(dedent`
+              The package ${colors.pink.bold(pkg)} is already installed and cannot be updated to ${colors.pink.bold(latestVersion)} because it would not satisfy "${colors.pink.bold(depRanges[pkg])}".
+            `);
+          }
+        }
+      }
+
+      if (hasInconsistentPackageVersions) {
+        reasons.push(
+          'Update your dependencies and try again, or manually install the Vitest addon.'
+        );
+      }
+    }
+
+    if (info.frameworkPackageName === '@storybook/nextjs') {
+      const nextVersion = await packageManager.getInstalledVersion('next');
+      if (!nextVersion) {
+        reasons.push(dedent`
+          It seems like you are using ${colors.pink.bold('@storybook/nextjs')} without having ${colors.pink.bold('next')} installed.
+          Please install "next" or use a different Storybook framework integration and try again.
+        `);
+      }
+    }
+
+    if (reasons.length > 0) {
+      reasons.unshift(
+        'The Test addon is incompatible with your current set up and cannot be installed:'
```

Comment:
The wording here (when I read it from the screenshot) indicates to me that this is not fixable, that fundamentally my project is just incompatible. As a user, I've reached a dead-end.

Only at the very end it somewhat indicates that I can maybe get it to work manually?
I'd indicate here that it's the _automatic_ workflow that doesn't work, but that it will still be doable manually. Maybe something like this:

```suggestion
        `The automated setup of the Test addon cannot currently succeed because of your project's incompatibilities listed below. Fear not, however, you can follow the manual installation process instead at ${c.cyan`https://storybook.js.org/docs/writing-tests/test-runner-with-vitest`}`
```

Reply:
Good idea!

Reply:
Nice idea but a little wordy. Trimmed down:

> Storybook Test's automated setup failed due to the following package incompatibilities. Fear not, however, you can follow the manual installation process instead at ${c.cyan`https://storybook.js.org/docs/writing-tests/test-runner-with-vitest`}